src/components/tables/CoinsTable.tsx

```diff
@@ -1,75 +1,231 @@
+    {
+      defaultColumn,
+      columns,
+      data: data,
+    },
+    useFilters,
+    useSortBy,
+    usePagination
+  )
+  const { pageIndex } = state
+
   return (
-    <table className="table">
-      <thead className="table-head">
-        <tr className="table-head-row">
-          <th>
-            <FormattedMessage
-              id="table.header.cell.name"
-              defaultMessage="NAME "
-            />
-          </th>
-          <th>
-            <FormattedMessage
-              id="table.header.cell.price"
-              defaultMessage="PRICE "
-            />
-          </th>
-          <th>
-            <FormattedMessage
-              id="table.header.cell.market.cap"
-              defaultMessage="MARKET CAP "
-            />
-          </th>
-          <th>
-            <FormattedMessage
-              id="table.header.cell.volume"
-              defaultMessage="VOLUME 24H"
-            />
-          </th>
-          <th>
-            <FormattedMessage
-              id="table.header.cell.low"
-              defaultMessage="LOW 24H  "
+    <>
+      <Box className="horizontal-scroll ">
+        <table {...getTableProps()}>
+          <thead className="table-head">
+            {headerGroups.map(headerGroup => (
+              <tr {...headerGroup.getHeaderGroupProps()}>
+                {headerGroup.headers.map(column => (
+                  <th {...column.getHeaderProps(column.getSortByToggleProps())}>
+                    {column.render('Header')}
+                    <span style={{ marginLeft: '10px' }}>
+                      {column.isSorted ? (
+                        column.isSortedDesc ? (
+                          <ArrowDownwardIcon sx={{ fontSize: '17px' }} />
+                        ) : (
+                          <ArrowUpwardIcon sx={{ fontSize: '17px' }} />
+                        )
+                      ) : (
+                        ''
+                      )}
+                    </span>
+                    <div>
+                      {column.canFilter ? column.render('Filter') : null}
+                    </div>
+                  </th>
+                ))}
+              </tr>
+            ))}
+          </thead>
+          <tbody {...getTableBodyProps()}>
+            {page.map(row => {
+              prepareRow(row)
+              return (
+                <tr {...row.getRowProps()}>
+                  {row.cells.map(cell => (
+                    <td {...cell.getCellProps()}>{cell.render('Cell')} </td>
+                  ))}
+                </tr>
+              )
+            })}
+          </tbody>
+        </table>
+      </Box>
+      <Box>
+        <Grid
+          container
+          justifyContent="space-between"
+          alignItems="flex-end"
+          height="80px"
+        >
+          <Grid item textAlign="left">
+            <TextField
+              size="small"
+              id="filled-basic"
+              label={intl.formatMessage({ id: 'table.search.page' })}
```

Comment:
Whenever you change a filter value, the page value should also reset. Otherwise I can be on page 6 and have 3 rows total in my table, which is not correct.